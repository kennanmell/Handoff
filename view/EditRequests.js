/**
 * Edit Requests view class, which allows organizations to see
 * previous requests and edit them.
 */

import React, { Component, PropTypes } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,
  TextInput,
  View,
  TouchableNativeFeedback,
  Modal,
  TouchableHighlight,
  Navigator
} from 'react-native';

/*Builds a EditRequests class, which an organization User uses to
select a request to edit and edit it in a pop-up. Has a RequestFeed
that the donor can interact with for more information in pop-ups.*/
export default class EditRequests extends Component {
  static propTypes = {
      onClose:PropTypes.func.isRequired,
  }

  constructor(props) {
    super(props);
    this.state = {text: '', modalVisible:false};
  }

  setModalVisible(visible) {
    this.setState({modalVisible: visible});
  }

  render() {
    return (

    <View style={{padding: 10}}>

       <TouchableHighlight style={styles.button} onPress={() => {
                 this.setModalVisible(true)
               }}><Text style={{color:'#FFFFFF'}}>Edit a Request</Text></TouchableHighlight>
        <Text style={styles.welcome}> Handoff </Text>
        <Text style={styles.instructions}> Request Feed Goes Here </Text>
        <Modal
                        animationType={"slide"}
                        transparent={false}
                        visible={this.state.modalVisible}
                        onRequestClose={() => {this.props.onClose}}
                >
                    <View style={{marginTop: 22}}>
                    <View>
                        <Text>Edit Request Here. Click Save When Done.</Text>
                        <TouchableHighlight save style={styles.button} onPress={() => { this.setModalVisible(!this.state.modalVisible)}}><Text style={{color:'#FFFFFF'}}>Save</Text></TouchableHighlight>
                        <TouchableHighlight close style={styles.button} onPress={this.props.onClose}> <Text style={{color:'#FFFFFF'}}>Cancel </Text></TouchableHighlight>
                    </View>
                    </View>
                </Modal>
  </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
  button: {
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'flex-end',
        marginLeft: 3,
        marginTop: 10,
        borderWidth: 1,
        padding: 5,
        borderColor: 'black',
        marginBottom: 15,
        backgroundColor: '#642D64',
    }
});