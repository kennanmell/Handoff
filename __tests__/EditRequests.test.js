import 'react-native';
import React from 'react';
import EditRequests from '../View/EditRequests';
import renderer from 'react-test-renderer';

// Ensures that the organization edit requests (EditRequests) renders correctly.
it('renders correctly', () => {
  const tree = renderer.create(
        <EditRequests
    		onClose={ () => {
    		}}
         />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});