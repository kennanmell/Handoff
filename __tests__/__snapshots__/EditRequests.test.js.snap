exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 10,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Edit a Request
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }>
     Handoff 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }>
     Request Feed Goes Here 
  </Text>
  <Modal
    animationType="slide"
    onRequestClose={[Function]}
    transparent={false}
    visible={false}>
    <View
      style={
        Object {
          "marginTop": 22,
        }
      }>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail">
          Edit Request Here. Click Save When Done.
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail">
            Save
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail">
            Cancel
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
